src/posit/connect/vanities.py

```diff
@@ -0,0 +1,236 @@
+
+    Vanities cannot have children. For example, if the vanity path "/finance/" exists, the vanity path "/finance/budget/" cannot. But, if "/finance" does not exist, both "/finance/budget/" and "/finance/report" are allowed.
+
+    The following vanities are reserved by Connect:
+    - `/__`
+    - `/favicon.ico`
+    - `/connect`
+    - `/apps`
+    - `/users`
+    - `/groups`
+    - `/setpassword`
+    - `/user-completion`
+    - `/confirm`
+    - `/recent`
+    - `/reports`
+    - `/plots`
+    - `/unpublished`
+    - `/settings`
+    - `/metrics`
+    - `/tokens`
+    - `/help`
+    - `/login`
+    - `/welcome`
+    - `/register`
+    - `/resetpassword`
+    - `/content`
+    """
+
+    AfterDestroyCallback = Callable[[], None]
+
+    class VanityAttributes(TypedDict):
+        """Vanity attributes."""
+
+        path: Required[str]
+        content_guid: Required[str]
+        created_time: Required[str]
+
+    def __init__(
+        self,
+        /,
+        params: ResourceParameters,
+        *,
+        after_destroy: Optional[AfterDestroyCallback] = None,
+        **kwargs: Unpack[VanityAttributes],
+    ):
+        """Initialize a Vanity.
+
+        Parameters
+        ----------
+        params : ResourceParameters
+        after_destroy : AfterDestroyCallback, optional
+            Called after the Vanity is successfully destroyed, by default None
+        """
+        super().__init__(params, **kwargs)
+        self._after_destroy = after_destroy
+        self._content_guid = kwargs["content_guid"]
+
+    @property
+    def _endpoint(self):
+        return self.params.url + f"v1/content/{self._content_guid}/vanity"
+
+    def destroy(self) -> None:
+        """Destroy the vanity.
+
+        Raises
+        ------
+        ValueError
+            If the foreign unique identifier is missing or its value is `None`.
+
+        Warnings
+        --------
+        This operation is irreversible.
+
+        Note
+        ----
+        This action requires administrator privileges.
+        """
+        self.params.session.delete(self._endpoint)
+
+        if self._after_destroy:
+            self._after_destroy()
+
+
+class Vanities(Resources):
+    """Manages a collection of vanities."""
+
+    def all(self) -> List[Vanity]:
+        """Retrieve all vanities.
+
+        Returns
+        -------
+        List[Vanity]
+
+        Notes
+        -----
+        This action requires administrator privileges.
+        """
+        endpoint = self.params.url + "v1/vanities"
+        response = self.params.session.get(endpoint)
+        results = response.json()
+        return [Vanity(self.params, **result) for result in results]
+
+
+class VanityMixin(Resource):
+    """Mixin class to add a vanity attribute to a resource."""
+
+    class HasGuid(TypedDict):
+        """Has a guid."""
+
+        guid: Required[str]
+
+    def __init__(self, /, params: ResourceParameters, **kwargs: Unpack[HasGuid]):
```

Comment:
What does `/` mean?

Reply:
https://docs.python.org/3/whatsnew/3.8.html#positional-only-parameters

Although, I no longer need this. It's cruft left over from implementation.

```suggestion
    def __init__(self, params: ResourceParameters, **kwargs: Unpack[HasGuid]):
```